ledger/src/blockstore.rs

```diff
@@ -3103,15 +3113,8 @@ impl Blockstore {
         &self,
         num: usize,
     ) -> Result<Vec<(Slot, Hash, UnixTimestamp)>> {
-        Ok(self
-            .db
-            .iter::<cf::OptimisticSlots>(IteratorMode::End)?
-            .take(num)
-            .map(|(slot, data)| {
-                let meta: OptimisticSlotMetaVersioned = deserialize(&data).unwrap();
-                (slot, meta.hash(), meta.timestamp())
-            })
-            .collect())
+        let iter = self.reversed_optimistic_slots_iterator()?;
+        Ok(iter.take(num).collect())
```

Comment:
moving the `take()` could incur extra work.  is that a concern?

Reply:
Hmm, are you sure? I thought functions that returned `impl Iterator ...` were lazy in evaluation. For example, consider this basic example that I believe illustrates this (noting that we only see 10 prints)
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c8e20e87931e0359f3516fa37b32ea71

Reply:
sure? no.  just got bit by `take_while()` working not at all the way i thought last week and figured it'd probably apply here too

Reply:
Ok, based on the playground link I provided, I think we're in the clear. If you want to send me the link or an example snippet of the `take_while()` bit you were referring to, I can take another look.